### DISCUSSION TITLE
Error when updating IP info

### DISCUSSION THREAD

**Discussion by charlessauer1** [netbox-community/netbox#12339] [Q&A]
Hi,

I have assigned IP's to devices and when updating the description field and press Enter the following error occurs of the specific user.

<class 'AttributeError'>
'NoneType' object has no attribute 'get_absolute_url'
Python version: 3.10.6
NetBox version: 3.3.8

What is the workaround in getting the error fixed and is this a bug or the way I'm working in entering info.

Charles

**Comment by charlessauer1** [comment]
I picked up a duplicated IP that was assigned but some of the assigned_object was not set. The problem is that is was not related to the IP that I updated, but seems like a validation process was triggered and error given.

**Reply by candlerb** [reply]
When you say "some of the assigned_object was not set", what do you mean?  Did it point to a non-existent object, or did it have assigned_object_type_id set but not assigned_object_id (or vice versa)?

There have been some changes recently in this area, see #11550 which causes problem #12301, which might be the same underlying cause.

**Comment by charlessauer1** [comment]
Hi Brian,

Below is a sample of what I saw, unfortunately the info from the API I didn't capture to identify.
IP Address
VRF
Status
Role
Tenant
Assigned
DNS Name
Description
ID
Tenant Group
NAT (Inside)
NAT (Outside)
Tags
Interface
Device/VM
10.3.1.90/24
Global
Active
10117
GP-xxxx-SW01
1107
MGMT
GP-xxxx-SW01
10.3.1.90/24
Global
Active
10117
GP-xxxx-SW01
1106

I will have a look at post you have made reference to.

Thanks

From: Brian Candler ***@***.***>
Sent: Tuesday, April 25, 2023 10:51 AM
To: netbox-community/netbox ***@***.***>
Cc: Charles Sauer ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [netbox-community/netbox] Error when updating IP info (Discussion #12339)


When you say "some of the assigned_object was not set", what do you mean? Did it point to a non-existent object, or did it have assigned_object_type_id set but not assigned_object_id (or vice versa)?

There have been some changes recently in this area, see #11550<https://github.com/netbox-community/netbox/pull/11550> which causes problem #12301<https://github.com/netbox-community/netbox/issues/12301>, which might be the same underlying cause.

-
Reply to this email directly, view it on GitHub<https://github.com/netbox-community/netbox/discussions/12339#discussioncomment-5717022>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A7NDVEC5BP22ROH6PNIL5WDXC6F7XANCNFSM6AAAAAAXKPWMIU>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

Disclaimer The information contained in this communication is confidential and may be legally privileged. It is intended solely for the use of ***@***.*** and others authorised to receive it. If you are not ***@***.*** you are hereby notified that any disclosure, copying, distribution or taking action in reliance of the contents of this information is strictly prohibited and may be unlawful. This e-mail and its contents are subject to the MetroFibre e-mail legal notice available at Disclaimer. <https://metrofibre.co.za/email/MetroFibre-Email-Disclaimer.pdf>

**Comment by charlessauer1** [comment]
Hi Brian,

The assigned_object_type was not set. Attach the scfeenshot of the Change Log indicating the duplicate.
![IP1](https://user-images.githubusercontent.com/131742352/234230727-43734d2b-eee3-47a2-ac76-83e225cbbac8.png)
![IP2](https://user-images.githubusercontent.com/131742352/234230760-81dc3d3a-585b-47d7-ba87-f99d5ea88315.png)

**Reply by candlerb** [reply]
And those are different objects? (I can't see their "id")

But yes, it looks like you've been caught by this problem.

You can run this custom script to look for and fix any other instances:
https://github.com/netbox-community/customizations/pull/85/files
(just copy this file to /opt/netbox/netbox/scripts/)

I suggest you run it with "commit changes" turned off first to find out how many it would touch.
